### ISSUE TITLE
Why does 'Unable to read' occur when check-disk-usage.rb is run as sensu user on Ubuntu xenial? 

### ISSUE THREAD

**Issue by 030** [sensu-plugins/sensu-plugins-disk-checks#53]
```
/opt/sensu/embedded/bin/check-disk-usage.rb 
CheckDisk OK: All disk usage under 85% and inode usage under 85%
```

running as sensu results in

```
sudo -u sensu /opt/sensu/embedded/bin/check-disk-usage.rb 
CheckDisk WARNING: /run/lxcfs/controllers Unable to read., /run/lxcfs/controllers/memory Unable to read., /run/lxcfs/controllers/cpuset Unable to read., /run/lxcfs/controllers/cpu,cpuacct Unable to read., /run/lxcfs/controllers/devices Unable to read., /run/lxcfs/controllers/perf_event Unable to read., /run/lxcfs/controllers/freezer Unable to read., /run/lxcfs/controllers/blkio Unable to read., /run/lxcfs/controllers/pids Unable to read., /run/lxcfs/controllers/net_cls,net_prio Unable to read., /run/lxcfs/controllers/hugetlb Unable to read., /run/lxcfs/controllers/name=systemd Unable to read.
```

**Comment by phumpal** [comment]
@030 I just ran into the same issue. This works but may not be correct depending on your needs

```
$ check-disk-usage.rb -w 75 -c 90 -x cgroup -p /run/lxcfs
CheckDisk OK: All disk usage under 75% and inode usage under 85%
```

In the Unit file sensu-client is configured to run as the sensu user and sensu group.

**Comment by 030** [comment]
In my opinion `check-disk-usage.rb` should work. No additional parameters should be required and the issue should never occur when just  `check-disk-usage.rb` is issued

**Comment by eyonov** [comment]
I'm having the exact same issue, but for anything under /home on CentOS. Why won't the sensu user be able to read the home directory? I don't want to exclude the whole /home directory as that would defeat the purpose of the check.

**Comment by vovkanaz** [comment]
"command": "/opt/sensu/embedded/bin/check-disk-usage.rb  -w 85 -c 95 -p /var/lib/docker /run/docker/netns/  /sys/kernel/debug/tracing  -x cgroup", or "command": "/opt/sensu/embedded/bin/check-disk-usage.rb  -w 85 -c 95 -p /var/lib/docker -p /sys/kernel/debug/tracing -p  /run/docker/netns/ -x cgroup"  also does not work when used docker on host but work when use only one parameter after -p.  It's bug or feature?

**Comment by awright424** [comment]
vovkanaz is right, is this a bug or a feature? I have multiple folders I need exlcuded.

**Comment by GhostLyrics** [comment]
@runningman84 have you tried `…usage.rb -x debugfs,tracefs`? Just asking because the first word in your output in the related line is `tracefs`, not `debugfs`.

**Comment by majormoses** [comment]
Should we add some documentation and call this one good?

**Comment by runningman84** [comment]
Yes but sane defaults would also help here. I guss nobody wants to check a tracefs or debugfs filesystem for disk usuage.

**Comment by GhostLyrics** [comment]
@runningman84 do you propose an internal whitelist?

**Comment by GhostLyrics** [comment]
I mean of course blacklist. Sorry.

**Comment by runningman84** [comment]
a blacklist yes

**Comment by majormoses** [comment]
@GhostLyrics when you say internal if you mean having this in a CONSTANT that is not accessible with an option I am not in favor. I am fine with using a constant to represent the majority of cases but think it is important to ensure that a user ALWAYS has control of these things as things change and it allows them to unblock themselves without waiting for a release.

**Comment by majormoses** [comment]
There is already the option `-i` to ignore mount points: https://github.com/sensu-plugins/sensu-plugins-disk-checks/blob/2.5.0/bin/check-disk-usage.rb#L46-L49. What I think people might want that does not exist yet is the ability to maybe ignore folders and not just mount points?

**Comment by RustyBower** [comment]
I think the issue is with something like docker, you have some randomly generated mount paths that need to be ignored. For example, I would love to be able to ignore: /sys/kernel/debug/tracing as well as the randomly generated docker ones /run/docker/netns/377f81fb80ce and /var/lib/docker/containers/blahblahblahblahblahblah/, so being able to ignore only full path for the multiple mount points is not really tenable. 

What would ideal is something like: 
check-disk-usage.rb -p /mnt/path1 -p /mnt/path2 -p /mnt/path3

I would even be fine with doing this in a comma seperated list like the current -i flag is

**Comment by majormoses** [comment]
If you want to submit a PR to allow `-p` to take a comma separated list I am :+1:

**Comment by RustyBower** [comment]
Will do! Time to dust off the old ruby skills...

**Comment by briangann** [comment]
this exclusion works for kubelet/docker on ubuntu 16.04: -x tmpfs,fuse,cgroup,overlay,nsfs,debugfs,tracefs,aufs

**Comment by prabhakarsultane** [comment]
> ```
> /opt/sensu/embedded/bin/check-disk-usage.rb 
> CheckDisk OK: All disk usage under 85% and inode usage under 85%
> ```
> running as sensu results in
> 
> ```
> sudo -u sensu /opt/sensu/embedded/bin/check-disk-usage.rb 
> CheckDisk WARNING: /run/lxcfs/controllers Unable to read., /run/lxcfs/controllers/memory Unable to read., /run/lxcfs/controllers/cpuset Unable to read., /run/lxcfs/controllers/cpu,cpuacct Unable to read., /run/lxcfs/controllers/devices Unable to read., /run/lxcfs/controllers/perf_event Unable to read., /run/lxcfs/controllers/freezer Unable to read., /run/lxcfs/controllers/blkio Unable to read., /run/lxcfs/controllers/pids Unable to read., /run/lxcfs/controllers/net_cls,net_prio Unable to read., /run/lxcfs/controllers/hugetlb Unable to read., /run/lxcfs/controllers/name=systemd Unable to read.
> ```

simply skip error like below 
./check-disk-usage.rb -p '(\/var|\/run|\/sys|\/snap)' -w 40 -c 50
CheckDisk WARNING: / 40.55% bytes usage (19.63 GiB/48.41 GiB)

**Comment by majormoses** [comment]
Does someone want to take a stab at writing something up in the readme? All the required functionality is present but I think people are struggling with the required options.

**Comment by phumpal** [comment]
@majormoses I think we overlooked your initial comment about the docs. Sorry!

I'll take a stab at it.

**Comment by Bhagyashri18** [comment]
@prabhakarsultane thanks, working.
